src/k2/module/k23si/Module.cpp

```diff
@@ -585,7 +585,7 @@ K23SIPartitionModule::handleQuery(dto::K23SIQueryRequest&& request, dto::K23SIQu
 
     IndexerIterator key_it = _initializeScan(request.key, request.reverseDirection, request.exclusiveKey);
     for (; !_isScanDone(key_it, request, response.results.size());
-                        _scanAdvance(key_it, request.reverseDirection, request.key.schemaName)) {
+                        _scanAdvance(key_it, request.reverseDirection, request.key.schemaName), numScans++) {
```

Comment:
I think it is better to move the increment into the for loop body, it is easy to miss here

Reply:
done.